pkg/scraper/client/resource/decode_test.go

```diff
@@ -241,3 +242,34 @@ container_start_time_seconds{container="metrics-server",namespace="kubernetes-da
 		})
 	}
 }
+
+func Fuzz_decodeBatch(f *testing.F) {
+	emptyMetrics := &storage.MetricsBatch{
+		Nodes: map[string]storage.MetricsPoint{},
+		Pods:  map[apitypes.NamespacedName]storage.PodMetricsPoint{},
+	}
+	f.Fuzz(func(t *testing.T, cpuValue float64, memValue float64, startTimeValue float64, timeStamp int64, defaultTimeValue int64) {
+		defaultTime := time.Unix(0, defaultTimeValue)
+		input := fmt.Sprintf(
+			`# HELP container_cpu_usage_seconds_total [ALPHA] Cumulative cpu time consumed by the container in core-seconds
+# TYPE container_cpu_usage_seconds_total counter
+container_cpu_usage_seconds_total{container="coredns",namespace="kube-system",pod="coredns-558bd4d5db-4dpjz"} %f %d
+# HELP container_memory_working_set_bytes [ALPHA] Current working set of the container in bytes
+# TYPE container_memory_working_set_bytes gauge
+container_memory_working_set_bytes{container="coredns",namespace="kube-system",pod="coredns-558bd4d5db-4dpjz"} %e %d
+# TYPE container_start_time_seconds gauge
+container_start_time_seconds{container="coredns",namespace="kube-system",pod="coredns-558bd4d5db-4dpjz"} %E %d`,
+			cpuValue, timeStamp, memValue, timeStamp, startTimeValue, timeStamp)
+		ms, err := decodeBatch([]byte(input), defaultTime, "node1")
+		switch {
+		case err != nil:
```

Comment:
Please don't use switch for error checking.

Reply:
ok